GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -0,0 +1,265 @@
+    );
+  });
+
+  //A file was given, but the resource doesn't exist
+  if (bitmapFontResourceName !== '' && !bitmapFontResource) {
+    console.warn(
+      'Unable to find bitmap font for resource "' +
+        bitmapFontResourceName +
+        '" (or is not a bitmapFont file).'
+    );
+  }
+
+  // There is no bitmap font file or resource
+  if (!bitmapFontResource || bitmapFontResourceName === '') {
+    // If the default Arial bitmap font doesn't exist, generate it and use it.
+    if (!PIXI.BitmapFont.available[this.getDefaultSlugFontName()]) {
+      return this.getDefaultBitmapFont();
+    }
+
+    return PIXI.BitmapFont.available[this.getDefaultSlugFontName()];
+  }
+
+  if (
+    this._loadedFontsData[bitmapFontResourceName].fontResourceFileName ===
+    bitmapFontResourceName
+  ) {
+    const fontData = this._loadedFontsData[bitmapFontResourceName].fontData;
+    const bitmapFont = this.patchInstalledBitmapFont(
+      PIXI.BitmapFont.install(fontData, texture)
+    );
+
+    this._loadedFontsData[bitmapFontResourceName].bitmapFont = bitmapFont;
+    this._loadedFontsData[bitmapFontResourceName].fontName = bitmapFont.font;
+    return this._loadedFontsData[bitmapFontResourceName].bitmapFont;
+  }
+};
+
+/**
+ * Check if a bitmap font exist by this name.
+ * The font name is defined in the font file (.fnt/.xml)
+ * @param {string} fontName Name of the bitmap font
+ */
+gdjs.BitmapFontManager.prototype.isBitmapFontLoaded = function (fontName) {
+  return !!PIXI.BitmapFont.available[fontName];
+};
+
+gdjs.BitmapFontManager.prototype.preloadBitmapFontData = async function (
+  onProgress,
+  onComplete
+) {
+  let resources = this._resources;
+
+  const bitmapFontResources = resources.filter(function (resource) {
+    return resource.kind === 'bitmapFont' && !resource.disablePreload;
+  });
+  if (bitmapFontResources.length === 0)
+    return onComplete(bitmapFontResources.length);
+
+  let loaded = 0;
+  /** @type BitmapFontManagerRequestCallback */
+  var onLoad = function (error) {
+    if (error) {
+      console.error('Error while preloading a bitmapFont resource:' + error);
+    }
+
+    loaded++;
+    if (loaded === bitmapFontResources.length) {
+      onComplete(bitmapFontResources.length);
+    } else {
+      onProgress(loaded, bitmapFontResources.length);
+    }
+  };
+
+  for (let i = 0; i < bitmapFontResources.length; ++i) {
+    const response = await fetch(bitmapFontResources[i].file)
+      .then(function (response) {
+        return response;
+      })
+      .catch(function (error) {
+        console.error(
+          "Can't fetch the bitmap font file " +
+            bitmapFontResources[i].file +
+            ', error: ' +
+            error
+        );
+      });
+
+    if (!response) return;
+
+    const fontData = await response.text();
+
+    this._loadedFontsData[bitmapFontResources[i].name] = {
+      fontResourceFileName: bitmapFontResources[i].name,
+      fontData: fontData,
+      fontName: '',
```

Comment:
Why are these empty strings?

Reply:
It is a reminder for knowing which values are in this._loadedFontsData[]

Reply:
Ok, and when are these values filled with some names?